src/archetype.rs

```diff
@@ -137,6 +137,13 @@ impl Archetype {
         &self.types
     }
 
+    /// Enumerate the types of the components of entities stored in this archetype
+    ///
+    /// Convenient for dispatching logic which needs to be performed on sets of type ids.
```

Comment:
Could we add some more specific discussion of when this is useful? A brief summary of a specific application or similar can be a big help for both helping other users discover good patterns, and for ensuring future refactoring preserves the desired capabilities.

Reply:
Done! I tried to describe the use case which motivated me to ask about this functionality, without getting into the specifics.